src/app/public/modules/data-manager/data-manager-ag-grid-adapter.directive.ts

```diff
@@ -0,0 +1,186 @@
+import {
+  AfterContentInit,
+  Directive,
+  ChangeDetectorRef,
+  ContentChildren,
+  QueryList,
+  Input
+} from '@angular/core';
+
+import {
+  Subscription
+} from 'rxjs';
+
+import {
+  AgGridAngular
+} from 'ag-grid-angular';
+
+import {
+  ColumnMovedEvent,
+  RowSelectedEvent
+} from 'ag-grid-community';
+
+import {
+  SkyDataManagerService
+} from './data-manager.service';
+
+import {
+  SkyDataViewConfig
+} from './models/data-view-config';
+
+import {
+  SkyDataManagerState
+} from './models/data-manager-state';
+
+import {
+  SkyDataManagerSortOption
+} from './models/data-manager-sort-option';
+
+@Directive({
+  selector: '[skyAgGridDataManager]'
+})
+export class SkyAgGridDataManagerAdapterDirective implements AfterContentInit {
+
+  @Input()
+  public viewConfig: SkyDataViewConfig;
+
+  private currentAgGrid: AgGridAngular;
+
+  @ContentChildren(AgGridAngular, { descendants: true })
+  public agGridList: QueryList<AgGridAngular>;
+
+  private dataStateSub: Subscription;
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private dataManagerSvc: SkyDataManagerService) { }
+
+  public ngAfterContentInit(): void {
+    this.checkForAgGrid();
+
+    this.agGridList.changes.subscribe(() => this.checkForAgGrid());
```

Comment:
There are several subscriptions in this file that don't get unsubscribed. Should we add a `takeUntil` to them?